File: reto02/reto02.py
def manufacture(gifts, materials):
    is_avaiable = True
    avaiables = []
    for gift in gifts:
        for char in gift:
            if char in materials:
                is_avaiable = True
            else:
                is_avaiable = False
                break
        if is_avaiable:
            avaiables.append(gift)
    return avaiables

File: reto02/reto02.js
function manufacture(gifts, materials) {
    let isAvailable = true;
    let availables = [];

    for (let i = 0; i < gifts.length; i++) {
        for (let j = 0; j < gifts[i].length; j++) {
            if (materials.includes(gifts[i][j])) {
                isAvailable = true;
            } else {
                isAvailable = false;
                break;
            }
        }

        if (isAvailable) {
            availables.push(gifts[i]);
        }
    }

    return availables;
}